src/Language/PureScript/Ide/Imports.hs

```diff
@@ -0,0 +1,193 @@
+import           Data.Monoid                        ((<>))
+import           Data.Text                          (Text)
+import qualified Data.Text                          as T
+import qualified Data.Text.IO                       as TIO
+import qualified Language.PureScript                as P
+import           Language.PureScript.Ide.Completion
+import           Language.PureScript.Ide.Error
+import           Language.PureScript.Ide.Filter
+import           Language.PureScript.Ide.State
+import           Language.PureScript.Ide.Types
+
+data Import = Import P.ModuleName P.ImportDeclarationType  (Maybe P.ModuleName)
+              deriving (Eq, Show)
+
+-- | Reads a file and returns the (lines before the imports, the imports, the
+-- lines after the imports)
+parseImportsFromFile :: (MonadIO m, MonadError PscIdeError m) =>
+                        FilePath -> m ([Text], [Import], [Text])
+parseImportsFromFile fp = do
+  file <- liftIO (TIO.readFile fp)
+  pure (sliceImportSection (T.lines file))
+
+sliceImportSection :: [Text] -> ([Text], [Import], [Text])
+sliceImportSection ls =
+  let
+    preImportSection = takeWhile (not . hasImportPrefix) ls
+    importSection =
+      takeWhile continuesImport $
+        dropWhile (not . hasImportPrefix) ls
+    postImportSection =
+      dropWhile continuesImport $
+        dropWhile (not . hasImportPrefix) ls
+    hasImportPrefix = T.isPrefixOf "import"
+    continuesImport x = hasImportPrefix x || T.isPrefixOf " " x || x == ""
+  in (preImportSection, parseImports importSection, postImportSection)
+
+
+-- | Concatenates multiline imports into a single line and tries to parse them.
+-- Anything that fails to parse gets left out
+parseImports :: [Text] -> [Import]
+parseImports ts =
+  let
+    concatMultilineImports = foldl step [] ts
+    step :: [Text] -> Text -> [Text]
+    step acc t = if T.isPrefixOf " " t
+                 then init acc ++ [last acc <> t]
+                 else acc ++ [t]
+  in
+    mapMaybe parseImport concatMultilineImports
+
+parseImport :: Text -> Maybe Import
+parseImport t =
+  case P.lex "<psc-ide>" (T.unpack t)
+       >>= P.runTokenParser "<psc-ide>" P.parseImportDeclaration' of
+    Right (mn, idt, mmn, _) -> Just (Import mn idt mmn)
+    Left _ -> Nothing
+
+
+-- | Adds an implicit import like @import Prelude@ to a Sourcefile.
+addImplicitImport :: (MonadIO m, MonadError PscIdeError m)
+                     => FilePath     -- ^ The Sourcefile read from
+                     -> P.ModuleName -- ^ The module to import
+                     -> m [Text]
+addImplicitImport fp mn = do
+  (pre, imports, post) <- parseImportsFromFile fp
+  let newImportSection = addImplicitImport' imports mn
+  pure $ pre
+    ++ newImportSection
+    ++ post
+
+addImplicitImport' :: [Import] -> P.ModuleName -> [Text]
+addImplicitImport' imports mn =
+  List.sort (map prettyPrintImport' (imports ++ [Import mn P.Implicit Nothing])) ++ [""]
```

Comment:
Sorting on strings might not be exactly what you want here, especially if `qualified` imports are still used.